addons/purchase/models/account_invoice.py

```diff
@@ -138,7 +138,13 @@ def invoice_line_move_line_get(self):
         if self.env.user.company_id.anglo_saxon_accounting:
             if self.type in ['in_invoice', 'in_refund']:
                 for i_line in self.invoice_line_ids:
-                    res.extend(self._anglo_saxon_purchase_move_lines(i_line, res))
+                    diff_line_list = self._anglo_saxon_purchase_move_lines(i_line, res)
+                    if diff_line_list: # This list always has max 1 element, as an invoice line can only match a single aml or none
+                        i_line_aml = [x for x in res if x.get('invl_id') == i_line['id']][0]
+                        diff_line = diff_line_list[0]
+                        i_line_aml['price_unit'] -= diff_line['price_unit']
+                        i_line_aml['price'] = i_line_aml['price_unit'] * i_line_aml['quantity']
+                        res.extend(diff_line_list)
```

Comment:
looks like it's wrong. I expect to receive the right lines directly from the function `_anglo_saxon_purchase_move_lines()`

okay I know it's messy, but previously `res` was changed in place directly in that function (see https://github.com/odoo/odoo/blob/saas-11.2/addons/purchase/models/account_invoice.py#L203)

I think, that line simply got removed by mistake... 

At least it removes the problem of finding back what was the account move line where the price difference needs to be deducted